src/main/java/pl/allegrotech/weatherapp/domain/Weather.java

```diff
@@ -0,0 +1,92 @@
+package pl.allegrotech.weatherapp.domain;
+
+import java.util.Objects;
+
+public final class Weather {
+
+    private final Location location;
+    private final Double temperature;
+
+    public Weather(Location location, Double temperature) {
+        this.location = location;
+        this.temperature = temperature;
+    }
+
+    public Location getLocation() {
+        return location;
+    }
+
+    public Double getTemperature() {
+        return temperature;
+    }
+
+    public WeatherDto toDto() {
+        var location = this.getLocation();
+        return new WeatherDto(
+                location.getLatitude(),
+                location.getLongitude(),
+                this.getTemperature()
+        );
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Weather weather = (Weather) o;
+        return Objects.equals(location, weather.location) && Objects.equals(temperature, weather.temperature);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(location, temperature);
+    }
+
+    @Override
+    public String toString() {
+        return "Weather{" +
+                "location=" + location +
+                ", temperature=" + temperature +
+                '}';
+    }
+
+    public static final class Location {
```

Comment:
Na pewno inner class to najlepszy pomysł w tym wypadku?

Reply:
Poprawione